src/AutocompleteBadges.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+import templates from './AutocompleteBadges.soy.js';
+import Autocomplete from 'metal-autocomplete';
+import Component from 'metal-component';
+import Soy from 'metal-soy';
+
+class AutocompleteBadges extends Component {
+
+	/**
+	 * @inheritDoc
+	 */
+	attached() {
+		this.autocomplete_ = new Autocomplete({
+			inputElement: this.refs.input,
+			data: this.getFilteredElements_.bind(this)
+		});		
+
+		this.autocomplete_.on('select', this.onListItemSelected_.bind(this));
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	rendered() {
```

Comment:
I understand why you've added this manual calculation for the input's width, but it's good to try to avoid changing dom elements directly when possible. I found a way that we can achieve this by just some changes in the html and css of this component 😃

This way we'll be able to remove both this method and `getFullWidthListBadges_` for good (and won't even need to have this `listBadges` ref either! yay!)

I'll explain this idea in the soy file itself, so take a look at my comment there.